src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2147,8 +2147,8 @@ class IndexColumnDefinitionSegment(BaseSegment):
     )
 
 
-@ansi_dialect.segment()
-class TableConstraintSegment(BaseSegment):
+@ansi_dialect.segmentTableConstraint()
+class Segment(BaseSegment):
```

Comment:
```suggestion
@ansi_dialect.segment()
class TableConstraintSegment(BaseSegment):
```
This shouldn't be changed